debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/memory/MemoryLogMinerEventProcessor.java

```diff
@@ -230,6 +230,15 @@ protected void addToTransaction(String transactionId, LogMinerEventRow row, Supp
                 getTransactionCache().put(transactionId, transaction);
             }
 
+            if (getConfig().getLogMiningBufferMemoryTransactionEventsThreshold() > 0
+                    && transaction.getEvents().size() >= getConfig().getLogMiningBufferMemoryTransactionEventsThreshold()) {
+                LOGGER.error("Transaction {} is oversized. Will abandon it.", transactionId);
+                // Discard the transaction from TransactionCache and to make it releasable.
+                getAndRemoveTransactionFromCache(transactionId).getEvents().clear();
+                abandonedTransactionsCache.add(transactionId);
+                metrics.incrementOversizedTransactions();
+                return;
+            }
```

Comment:
Let's pull this up to the abstract parent class & hook into that behavior in all impls of the processors.